src/ims/store/_db.py

```diff
@@ -1523,12 +1559,15 @@ async def incidentReports(self, eventID: str) -> Iterable[IncidentReport]:
         """
 
         def incidentReports(txn: Transaction) -> Iterable[IncidentReport]:
-            return tuple(
-                self._fetchIncidentReport(eventID, number, txn)
-                for number in tuple(
-                    self._fetchIncidentReportNumbers(eventID, txn)
-                )
```

Comment:
this is where we currently do thousands of queries anytime anytime the web client asks for all incident reports